src/contracts/GPv2Settlement.sol

```diff
@@ -300,6 +300,22 @@ contract GPv2Settlement {
         filledAmount[uidKey] = currentFilledAmount;
     }
 
+    /// @dev Allows settlment function to make arbitrary contract executions.
+    /// @param interaction contains address and calldata of the contract interaction.
+    function executeInteraction(GPv2Encoding.Interaction memory interaction)
+        internal
+    {
+        // To prevent possible attack on user funds, we explicitly disable
+        // interactions with AllowanceManager contract.
+        require(
+            interaction.target != address(allowanceManager),
+            "GPv2: forbidden interaction"
+        );
+        // solhint-disable-next-line avoid-low-level-calls
+        (bool success, ) = (interaction.target).call(interaction.callData);
+        require(success, "GPv2: failed interaction");
```

Comment:
Do you know if Etherscan (or some other tool) allows you to read the failure string of a failed internal call? It would be nice to know exactly what failed. Is it worth to print the error string here instead of our own custom message?

Reply:
Right, I suspect the error string is contained in the second parameter of the tuple being returned:

So if I did
```
(bool success, bytes memory response) = (interaction.target).call(interaction.callData)
```

Instead then I would have access to the response which may include error string when there is an error. I am not sure we want to use response in place of the current string since we would be unable to test... 

However, with the current implementation, I could try and see if we get "I always revert!" back.

Reply:
Nah, I can't get the response as a string because it comes back as bytes.

Reply:
Actually YES, I can do this:
```
        (bool success, bytes memory response) =
            (interaction.target).call(interaction.callData);
        require(success, string(response));
```

Reply:
I will include this in a commit, but not sure if it will stick